autoPyTorch/data/tabular_feature_validator.py

```diff
@@ -264,6 +265,17 @@ def transform(
 
         if hasattr(X, "iloc") and not scipy.sparse.issparse(X):
             X = cast(Type[pd.DataFrame], X)
+            if self.all_nan_columns is not None:
+                for column in X.columns:
+                    if column in self.all_nan_columns:
+                        if not X[column].isna().all():
+                            X[column] = np.nan
+                        X[column] = pd.to_numeric(X[column])
+            if len(self.categorical_columns) > 0:
+                categorical_columns = self.column_transformer.transformers_[0][-1]
+                for column in categorical_columns:
+                    if X[column].isna().all():
+                        X[column] = X[column].astype('object')
```

Comment:
Can we just do the following here because `hasattr(X, "iloc") and not scipy.sparse.issparse(X)` is the exact condition that we will get `self.all_nan_columns is not None`?

```suggestion
            if self.all_nan_columns is None:
                raise ValueError('_fit must be called before calling transform')

            for col in list(self.all_nan_columns):
                X[col] = np.nan
                X[col] = pd.to_numeric(X[col])

            if len(self.categorical_columns) > 0:
                categorical_cols = self.column_transformer.transformers_[0][-1]
                for col in categorical_cols:
                    if X[col].isna().all():  # all nan yields numeric, so modify it here
                        X[col] = X[col].astype('object')
```

Reply:
it will be none for sparse data. Again, we would like to keep it the same as development right now.

Reply:
Ok, then I will make a new PR only for this change as it is way much quicker (around 100~1000x).

Reply:
Hey, did you mean reg-cocktail dev?
I did not see this process in the dev branch.